include/solvers/heat_transfer.h

```diff
@@ -127,6 +127,12 @@ class HeatTransfer : public AuxiliaryPhysics<dim, TrilinosWrappers::MPI::Vector>
   void
   assemble_rhs();
 
+  /**
+   * @brief Call for the assembly of the matrix and the right-hand side of the Nitsche restriction for the heat transfert equation.
+   */
+  void
+  assemble_nitsche_restriction(bool assemble_matrix);
```

Comment:
From the header I get that it's only for the heat restriction, I think a better name for this function would be `assemble_nitsche_heat_restriction()`